docs/tutorial.md

```diff
@@ -0,0 +1,372 @@
+              of an object. Servers should convert recognized schemas to the latest
+              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            type: string
+          kind:
+            description: 'Kind is a string value representing the REST resource this
+              object represents. Servers may infer this from the endpoint the client
+              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: MemcachedSpec defines the desired state of Memcached
+            properties:
+              size:
+                description: Size defines the number of Memcached instances
+                format: int32
+                type: integer
+            type: object
+          status:
+            description: MemcachedStatus defines the observed state of Memcached
+            properties:
+              nodes:
+                description: Nodes store the name of the pods which are running Memcached
+                  instances
+                items:
+                  type: string
+                type: array
+            type: object
+        type: object
+    served: true
+    storage: true
+    subresources:
+      status: {}
+status:
+  acceptedNames:
+    kind: ""
+    plural: ""
+  conditions: []
+  storedVersions: []
+```
+
+### Custom Resource - memcached-sample.yaml
+
+```
+apiVersion: cache.example.com/v1alpha1
+kind: Memcached
+metadata:
+  name: memcached-sample
+spec:
+  # Add fields here
+  size: 1
+```
+
+## Implement the Controller
+
+**Note**: The next two subsections explain how the two methods `createOrUpdateResource` and `deleteResource`. These two methods will be called whenever there is some update/create/delete event occurs in the cluster.
+
+These methods are already scaffolded as part of `create api` command.
+
+### createOrUpdateResource
+
+This methid inaitlly checks for the deployment by using below code.
+
+```
+        Deployment deployment =
+                client
+                        .apps()
+                        .deployments()
+                        .inNamespace(resource.getMetadata().getNamespace())
+                        .withName(resource.getMetadata().getName())
+                        .get();
+```
+
+If the deployment is `null` that means we need to create deployment for it.
+
+Below code will check is deployment is null or not. If it is null then it will create deployment. `createMemcachedDeployment(resource)` creates the Deployment and then it will be applied by using `client.apps().deployments().create(newDeployment);` code. `createMemcachedDeployment(resource)` method explained in the next part.
```

Comment:
check is deployment is null? rephrase this

Reply:
Modified the description.